### PR TITLE
Migration generation

### PR DESCRIPTION
**Pull request by josegonzalez**
This is my first pass at making migration baking a little easier to use.

- Moved the existing bake behavior to a `--snapshot` option
- Moved the template to `migration.ctp` to `snapshot.ctp`
- Cleaned up the actual bake task a bit
- Added a `MigrationHelper` class to handle data output for migration files

This PR handles the signed/unsigned bug, and adds support for more phinx-supported attributes.

TODO:

- [x] Update docs
- [x] Rails-style migration generation
- [x] Test bake output the same way that the core does
- [x] Add tests for the `MigrationHelper` class

### CODE REVIEW

**Comment by josegonzalez** [comment]
I'm having a bit of troubling getting fixtures to run. I'm going to push my current code in the hopes that someone can figure out why fixtures aren't being inserted into the db...

**Comment by jadb** [comment]
@josegonzalez I'll try it some time today or tomorrow and share what I find.

**Comment by josegonzalez** [comment]
I ended up splitting out the snapshot generation and generation from the cli into two different tasks. I might still consolidate them to keep the interface the same, but I like how it came out :)

File: .travis.yml [hunk 1]
```diff
@@ -18,8 +17,8 @@ env:
 matrix:
   allow_failures:
     - php: hhvm
-
     - php: hhvm-nightly
+    - env: DB=pgsql db_dsn='postgres://postgres@127.0.0.1/cakephp_test'
```
Comment (lorenzo) [inline .travis.yml]: why?

File: src/View/Helper/MigrationHelper.php [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+    }
+
+    /**
+     * Returns an array converted into a formatted multiline string
+     *
+     * @param array $list array of items to be stringified
+     * @param array $options options to use
+     * @return string
+     */
+    public function stringifyList(array $list, array $options = [])
```
Comment (lorenzo) [inline src/View/Helper/MigrationHelper.php]: Isn't this part of the bake helper?

File: src/Shell/Task/SimpleMigrationTask.php [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function bake($name)
+    {
+        $this->params['no-test'] = true;
+        $collection = $this->getCollection($this->connection);
+        EventManager::instance()->attach(function (Event $event) use ($collection) {
```
Comment (lorenzo) [inline src/Shell/Task/SimpleMigrationTask.php]: `on` should be used instead of `attach`

**Comment by lorenzo** [comment]
This looks great! :+1:

File: .travis.yml (hunk 1 above)
Comment (josegonzalez) [inline .travis.yml]: Why what?

Comment (josegonzalez) [inline .travis.yml]: Oh, thats related to a bug in the schema generation in CakePHP. See [this commit](https://github.com/cakephp/migrations/commit/a57b25e5e3ac6d144441f4ebba0c0d4f80c46275) for more details.

File: src/View/Helper/MigrationHelper.php (hunk 2 above)
Comment (josegonzalez) [inline src/View/Helper/MigrationHelper.php]: The one in there doesn't properly handle value output (null, bools, integers) and the phinx fails *loudly* on **bad** input

File: src/Shell/Task/SimpleMigrationTask.php (hunk 3 above)
Comment (josegonzalez) [inline src/Shell/Task/SimpleMigrationTask.php]: I'll take a look at updating this one soon.

File: src/View/Helper/MigrationHelper.php (hunk 2 above)
Comment (AD7six) [inline src/View/Helper/MigrationHelper.php]: Why not fix bake's helper? If you can give me/explain a failing use case I can take care of that.

File: tests/TestCase/Util/ColumnParserTest.php [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+        $this->assertEquals([
+            'id' => [
+                'columnType' => 'integer',
+                'options' => [
+                    'null' => false,
+                    'default' => null,
+                    'limit' => 11,
+                ],
+            ],
+        ], $this->columnParser->parseFields(['id:primary']));
```
Comment (markstory) [inline tests/TestCase/Util/ColumnParserTest.php]: Some troll is going to want `:` in their column names.

File: tests/TestCase/View/Helper/MigrationHelperTest.php [hunk 5]
```diff
@@ -0,0 +1,348 @@
...
+            'key' => 'value',
+        ]));
+        $this->assertEquals("
+        'key' => 'value',
+        'other_key' => 'other_value',
+    ", $this->Helper->stringifyList([
+            'key' => 'value',
+            'other_key' => 'other_value',
+        ]));
+        $this->assertEquals("
```
Comment (markstory) [inline tests/TestCase/View/Helper/MigrationHelperTest.php]: indent.

**Comment by markstory** [comment]
Looks great :+1:

File: tests/TestCase/Util/ColumnParserTest.php (hunk 4 above)
Comment (josegonzalez) [inline tests/TestCase/Util/ColumnParserTest.php]: That troll can write his own migrations sans the cli-based generator.

Comment (jadb) [inline tests/TestCase/Util/ColumnParserTest.php]: :100:

**Pull request merged**